include/unifex/thread_unsafe_event_loop.hpp

```diff
@@ -64,158 +65,180 @@ class thread_unsafe_event_loop {
+
+    template <typename Receiver>
+    after_operation<Duration, std::remove_cvref_t<Receiver>> connect(Receiver&& r) && {
+      return after_operation<Duration, std::remove_cvref_t<Receiver>>{
+          (Receiver &&) r, duration_, loop_};
+    }
+   private:
+    friend scheduler;
 
-      thread_unsafe_event_loop& loop_;
-      Duration duration_;
-    };
+    explicit type(
+        thread_unsafe_event_loop& loop,
+        Duration duration) noexcept
+        : loop_(loop), duration_(duration) {}
 
-    struct schedule_at_sender {
-      template <
-          template <typename...> class Variant,
-          template <typename...> class Tuple>
-      using value_types = Variant<Tuple<>>;
+    thread_unsafe_event_loop& loop_;
+    Duration duration_;
+  };
 
-      template <template <typename...> class Variant>
-      using error_types = Variant<>;
+  struct schedule_at_sender;
 
-     private:
-      template <typename Receiver>
-      class operation final : public operation_base {
-       public:
-        void start() noexcept {
-          callback_.construct(
-              get_stop_token(receiver_), cancel_callback{*this});
-          operation_base::start();
-        }
+  template <typename Receiver>
+  struct at_op_ {
+    class type;
+  };
+  template <typename Receiver>
+  using at_operation = typename at_op_<std::remove_cvref_t<Receiver>>::type;
 
-       private:
-        friend schedule_at_sender;
+  template <typename Receiver>
+  class at_op_<Receiver>::type final : public operation_base {
+   public:
+    void start() noexcept {
+      callback_.construct(
+          get_stop_token(receiver_), cancel_callback{*this});
+      operation_base::start();
+    }
 
-        template <typename Receiver2>
-        explicit operation(
-            Receiver2&& r,
-            time_point_t tp,
-            thread_unsafe_event_loop& loop)
-            : operation_base(loop), receiver_((Receiver2 &&) r) {
-          this->dueTime_ = tp;
-        }
+   private:
+    friend schedule_at_sender;
+
+    template <typename Receiver2>
+    explicit type(
+        Receiver2&& r,
+        time_point_t tp,
+        thread_unsafe_event_loop& loop)
+        : operation_base(loop), receiver_((Receiver2 &&) r) {
+      this->dueTime_ = tp;
+    }
 
-        void execute() noexcept override {
-          callback_.destruct();
-          if constexpr (is_stop_never_possible_v<
-                            stop_token_type_t<Receiver&>>) {
-            unifex::set_value(std::move(receiver_));
-          } else {
-            if (get_stop_token(receiver_).stop_requested()) {
-              unifex::set_done(std::move(receiver_));
-            } else {
-              unifex::set_value(std::move(receiver_));
-            }
-          }
+    void execute() noexcept override {
+      callback_.destruct();
+      if constexpr (is_stop_never_possible_v<
+                        stop_token_type_t<Receiver&>>) {
+        unifex::set_value(std::move(receiver_));
```

Comment:
I just noticed that this call to set_value() could potentially be throwing (since we don't require that set_value() customisations are noexcept any more).

We probably need to do a pass through libunifex to pick up calls to set_value() and wrap them in try/catch if they are not noexcept.

Reply:
I'll leave it for now.